### DISCUSSION TITLE
How to share YAML front matter across documents?

### DISCUSSION THREAD

**Discussion by lillemets** [quarto-dev/quarto-cli#2610] [Q&A]
I have a directory containing Revealjs presentations formatted as .qmd documents. Currently each file has a 12 lines of YAML front matter of which only one line (title) is not identical across documents. With Rmarkdown I could just put the execution options in `_outupt.yml`. I attempted the same with `_quarto.yml` but that is completely ignored when rendering Revealjs presentations in Rstudio.

How could I mimic the `_output.yml` of Rmarkdown in Quarto?

**Comment by dragonstyle** [comment]
You have a couple of options:

1) Use a project file <https://quarto.org/docs/projects/quarto-projects.html#project-metadata>

2) Use a `_metadata.yml` file within a directory to apply metadata to the files in the directory (this seems like what you're looking for assuming you don't want to group the revealjs presentations into a project) <https://quarto.org/docs/projects/quarto-projects.html#project-metadata>

HTH!

**Comment by lillemets** [comment]
I have read the referred instructions but as I mentioned, `_quarto.yml` is simply ignored when rendering.

In the `.qmd` YAML I have 

```
---
title: Title
---
```

In the same directory as the `.qmd` file I can have `_quarto.yml` or `_metadata.yml` containing

```
format: 
  revealjs: 
    self-contained: true
    preview-links: auto
    slide-number: c/t
```

The result of rendering the `.qmd` is a regular HTML file and what ever parameters I set are ignored. However, setting `execute-dir: project` in `_quarto.yml` seems to work. So it is just the execution options that are ignored in all `.yml` files. I must be doing something wrong but I have no idea what it is.

**Reply by dragonstyle** [reply]
I'm not seeing this behavior as I test locally. Using the above examples (I added a reveal theme to make it very obvious) with the metadata in a `_quarto.yml` file next to the input file resulted in a properly themed reveal presentation when rendering. I was testing either in VS.Code or just using the terminal directly (perhaps our tooling might account for the difference).

My original answer is pretty confusing/wrong about `_metadata.yml` files as these only work in the context of projects (which is what establishes the boundaries of containership where we'll search for `_metadata` files. You really want the project in this case, unless this folder is already a subfolder of a project...

**Reply by lillemets** [reply]
I really appreciate your effort. I guess I'm not completely wrong then. I do have `_metadata.yml` in the same directory as the `.qmd` and `_quarto.yml` and `.quarto` in parent directory. As said, neither of the YAML files are recognized when rendering.

It might have something to do with RStudio. I'm currently unable to test rendering in terminal as the Quarto package has dependency issues on Arch Linux but will try that when possible.
